crates/rspack_plugin_css/src/parser_and_generator/mod.rs

```diff
@@ -404,11 +442,33 @@ impl ParserAndGenerator for CssParserAndGenerator {
 
         if let Some(exports) = &self.exports {
           let mg = compilation.get_module_graph();
-          let unused = get_unused_local_ident(exports, identifier, generate_context.runtime, &mg);
-          context.data.insert(unused);
 
           let used = get_used_exports(exports, identifier, generate_context.runtime, &mg);
 
+          let mut deps = HashSet::new();
+          let mut todo: Vec<_> = used.values().map(|v| *v).collect();
+
+          let mut idx = 0;
+
+          while idx < todo.len() {
+            let next = todo[idx];
+
+            for exp in next {
+              if deps.insert(exp.ident.clone()) {
+                if let Some(new) = exports.get(&exp.ident) {
+                  todo.push(new)
+                }
+              }
+            }
+
+            idx += 1;
+          }
+
+          let unused =
+            get_unused_local_ident(exports, identifier, generate_context.runtime, &mg, deps);
+
+          context.data.insert(unused);
+
```

Comment:
As a temporary workaround, we could removed deps from unused here mannually, so `composes` would work. However it is still impossible to make keyframe work because we don't know the classes they're in. What would be the proper fix? @ahabhgk